graph/simple_graph.py

```diff
@@ -0,0 +1,71 @@
+class Graph(object):
+    """Implements a graph data structure"""
+
+    def __init__(self):
+        self.graph_dict = {}
+
+    def nodes(self):
+        """return a list of all nodes in the graph"""
+        return self.graph_dict.keys()
+
+    def edges(self):
+        """return a list of all edges in the graph"""
+
+        return [(key, node) for key, value in
+                self.graph_dict.iteritems() for node in value]
+
+    def add_node(self, node):
+        """add a new node 'n' to the graph"""
+        self.graph_dict[node] = []
```

Comment:
What happens if node is already in the graph with some edges leading from it?  Write a test that shows the failure, then fix it.